src/strategies/python.ts

```diff
@@ -148,6 +167,35 @@ export class Python extends BaseStrategy {
     }
   }
 
+  protected async getArtifactName(): Promise<string | null> {
+    const ARTIFACT_NAME_REGEX = /name *= *['"](?<name>.*)['"](\r|\n|$)/;
+    const setupPyContents = await this.getSetupPyContents();
+    if (setupPyContents) {
+      const match = setupPyContents.match(ARTIFACT_NAME_REGEX);
+      if (match && match?.groups?.name) {
+        return match.groups.name;
+      }
+    }
+    return null;
+  }
```

Comment:
This strategy actually supports several types of python projects. There are other files besides setup.py that can specify the package name (see the buildUpdates() for the others)

Reply:
Switched logic to 

`const artifactName = projectName ?? (await this.getNameFromSetupPy());`

If an artifact name is provided to the constructor, or found in a `pyproject.toml`, it will be used.